lib/spack/spack/cmd/config.py

```diff
@@ -43,29 +50,81 @@ def setup_parser(subparser):
                              help="configuration section to edit. "
                                   "options: %(choices)s",
                              metavar='SECTION',
+                             nargs='?',
                              choices=spack.config.section_schemas)
+    edit_parser.add_argument(
+        '--print-file', action='store_true',
+        help="print the file name that would be edited")
+
+
+def _get_scope_and_section(args):
+    """Extract config scope and section from arguments."""
+    scope = args.scope
+    section = args.section
+
+    # w/no args and an active environment, point to env manifest
+    if not args.section:
+        env = ev.get_env(args, 'config edit', required=False)
+        if env:
+            scope = env.env_file_config_scope_name()
+
+    # set scope defaults
+    elif not args.scope:
+        if section == 'compilers':
+            scope = spack.config.default_modify_scope()
+        else:
+            scope = 'user'
+
+    return scope, section
 
 
 def config_get(args):
-    spack.config.config.print_section(args.section)
+    """Dump merged YAML configuration for a specific section.
+
+    With no arguments and an active environment, print the contents of
+    the environment's manifest file (spack.yaml).
+
+    """
+    scope, section = _get_scope_and_section(args)
+
+    if scope and scope.startswith('env:'):
+        config_file = spack.config.config.get_config_filename(scope, section)
+        if os.path.exists(config_file):
+            with open(config_file) as f:
+                print(f.read())
+        else:
+            tty.die('environment has no %s file' % ev.manifest_name)
+
+    elif section is not None:
+        spack.config.config.print_section(section)
+
+    else:
+        tty.die('`spack config get` requires a section argument '
+                'or an active environment.')
 
 
 def config_blame(args):
+    """Print out line-by-line blame of merged YAML."""
     spack.config.config.print_section(args.section, blame=True)
 
 
 def config_edit(args):
-    if not args.scope:
-        if args.section == 'compilers':
-            args.scope = spack.config.default_modify_scope()
-        else:
-            args.scope = 'user'
-    if not args.section:
-        args.section = None
-
-    config = spack.config.config
-    config_file = config.get_config_filename(args.scope, args.section)
-    editor(config_file)
+    """Edit the configuration file for a specific scope and config section.
+
+    With no arguments and an active environment, edit the spack.yaml for
+    the active environment.
+
+    """
+    scope, section = _get_scope_and_section(args)
+    if not scope and not section:
```

Comment:
I agree this is confusing.  `section` is the required argument (`compilers`, `repos`).  `scope` is optional.  If you edit a specific scope and have no section, we currently throw an error because you may have specified a scope that has multiple config section files in it.  If you specify a section but not a scope we take the default modify scope.  So it's really `and`.